addons/mrp_repair/tests/test_flows.py

```diff
@@ -0,0 +1,50 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import Form, tagged
+from odoo.addons.mrp.tests.common import TestMrpCommon
+from odoo.exceptions import AccessError
+
+
+@tagged('post_install', '-at_install')
+class TestMrpRepairFlows(TestMrpCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env.ref('base.group_user').write({'implied_ids': [(4, cls.env.ref('stock.group_production_lot').id)]})
+
+    def test_possible_to_add_kit_after_confirm(self):
+        """
+        Test that it is possible to add a kit manufactured product to an already confirmed Repair Order
+        """
+        repaired = self.env['product.product'].create({'name': 'Repaired'})
+        part = self.env['product.product'].create({'name': 'Kit Component'})
+        kit = self.env['product.product'].create({
+            'name': 'Kit',
+            'type': 'product',
+        })
+        bom = self.env['mrp.bom'].create({
+            'product_id': kit.id,
+            'product_tmpl_id': kit.product_tmpl_id.id,
+            'type': 'phantom',
+        })
+        bom.write({
+            'bom_line_ids': [self.env['mrp.bom.line'].create({
```

Comment:
Why not directly add the line while creating the BoM ?
We could then avoid a useless call to `write`
Also, be careful when dealing with `2many` fields, you need to use some [commands](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.fields.Command)
```suggestion
            'bom_line_ids': [(0, 0, {
```